chains/evm/calls/bridge/bridge.go

```diff
@@ -0,0 +1,414 @@
+	withdrawInput, err := c.PrepareWithdrawInput(
+		handlerAddress, tokenAddress, recipientAddress, amountOrTokenId,
+	)
+	if err != nil {
+		log.Error().Err(err)
+		return nil, err
+	}
+	return c.executeTransaction(withdrawInput, opts, "Withdraw")
+}
+
+func (c *BridgeContract) GetThreshold() (uint8, error) {
+	input, err := c.PackMethod("_relayerThreshold")
+	if err != nil {
+		return 0, err
+	}
+	msg := ethereum.CallMsg{From: common.Address{}, To: &c.bridgeContractAddress, Data: input}
+	out, err := c.client.CallContract(context.TODO(), client.ToCallArg(msg), nil)
+	if err != nil {
+		return 0, err
+	}
+	res, err := c.UnpackResult("_relayerThreshold", out)
+	if err != nil {
+		return 0, err
+	}
+	out0 := *abi.ConvertType(res[0], new(uint8)).(*uint8)
+	return out0, nil
+}
+
+func (c *BridgeContract) IsRelayer(relayerAddress common.Address) (bool, error) {
+	input, err := c.PrepareIsRelayerInput(relayerAddress)
+	if err != nil {
+		return false, err
+	}
+
+	msg := ethereum.CallMsg{
+		From: common.Address{},
+		To:   &c.bridgeContractAddress,
+		Data: input,
+	}
+	out, err := c.client.CallContract(context.TODO(), client.ToCallArg(msg), nil)
+	if err != nil {
+		return false, err
+	}
+
+	if len(out) == 0 {
+		// Make sure we have a contract to operate on, and bail out otherwise.
+		if code, err := c.client.CodeAt(context.Background(), c.bridgeContractAddress, nil); err != nil {
+			return false, err
+		} else if len(code) == 0 {
+			return false, fmt.Errorf("no code at provided address %s", c.bridgeContractAddress.String())
+		}
+	}
+
+	return c.ParseIsRelayerOutput(out)
+}
+
+func (c *BridgeContract) ProposalStatus(p *proposal.Proposal) (message.ProposalStatus, error) {
+	input, err := c.PackMethod("getProposal", p.Source, p.DepositNonce, p.GetDataHash())
+	if err != nil {
+		return message.ProposalStatus{}, err
+	}
+	msg := ethereum.CallMsg{From: common.Address{}, To: &p.BridgeAddress, Data: input}
+	out, err := c.client.CallContract(context.TODO(), client.ToCallArg(msg), nil)
+	if err != nil {
+		return message.ProposalStatus{}, err
+	}
+	res, err := c.UnpackResult("getProposal", out)
+	if err != nil {
+		return message.ProposalStatus{}, err
+	}
+	ps := *abi.ConvertType(res[0], new(message.ProposalStatus)).(*message.ProposalStatus)
+	return ps, nil
+}
+
+func (c *BridgeContract) IsProposalVotedBy(by common.Address, p *proposal.Proposal) (bool, error) {
+	input, err := c.PackMethod("_hasVotedOnProposal", idAndNonce(p.Source, p.DepositNonce), p.GetDataHash(), by)
+	if err != nil {
+		return false, err
+	}
+	msg := ethereum.CallMsg{From: common.Address{}, To: &p.BridgeAddress, Data: input}
+	out, err := c.client.CallContract(context.TODO(), client.ToCallArg(msg), nil)
+	if err != nil {
+		return false, err
+	}
+	res, err := c.UnpackResult("_hasVotedOnProposal", out)
+	if err != nil {
+		return false, err
+	}
+	out0 := *abi.ConvertType(res[0], new(bool)).(*bool)
+	return out0, nil
+}
+
+func (c *BridgeContract) executeTransaction(
```

Comment:
Not sure if this function is overengineering - just wanted to make `bridge.go` more manageable to read.

Another idea is to move all these `Prepare....` functions to separate file `prepare.go` so it is easier to navigate through this file. What do you think?

Reply:
Separating data preparation for execution (transacting) looks as good idea for me. We are following Single responsibility principle by this, that allows us to write unit tests for prepare easier. 
Yeah preparation should be moved to separate file then. And probably they can be made non import